### PR TITLE
#26: split file ast by module

### PR DESCRIPTION
**Pull request by pbrudnick**

### CODE REVIEW

File: lib/meandro/util.ex [hunk 1]
```diff
@@ -16,27 +16,46 @@ defmodule Meandro.Util do
   It can be in `:parallel` or `:sequential` depending its second argument.
   """
   @spec parse_files([Path.t()], parsing_style()) :: [
-          {Path.t(), Macro.t()}
+          {Path.t(), [{atom(), Macro.t()}]}
```
Comment (elbrujohalcon) [inline lib/meandro/util.ex:19]:
```suggestion
          {Path.t(), [{module(), Macro.t()}]}
```

**Review by tak30 — CHANGES_REQUESTED** [review]
We need to test it also for modules inside modules

File: lib/meandro/util.ex [hunk 2]
```diff
@@ -16,27 +16,47 @@ defmodule Meandro.Util do
...
+    |> maybe_split_by_module(file)
+  end
+
+  defp maybe_split_by_module(ast, file) do
+    {_, result} = Macro.prewalk(ast, [], &collect_modules/2)
+    {file, result}
+  end
+
+  defp collect_modules({:defmodule, _, _} = module_node, acc) do
+    {module_node, [{module_name(module_node), module_node} | acc]}
```
Comment (tak30) [inline lib/meandro/util.ex:48]: I believe we should also pop out nested module from the ast in order to not analyze them twice

**Review by tak30 — CHANGES_REQUESTED** [review]

File: lib/meandro/util.ex (hunk 2 above)
Comment (maco) [reply lib/meandro/util.ex:48]: agreed

**Review by maco — CHANGES_REQUESTED** [review]

File: .tool-versions [hunk 3]
```diff
@@ -0,0 +1,2 @@
+elixir 1.13.4
+erlang 24.3.3
```
Comment (maco) [inline .tool-versions:2]: probably best not to include this file in a library

File: lib/meandro/util.ex [hunk 4]
```diff
@@ -33,8 +30,7 @@ defmodule Meandro.Util do
 
   defp file_to_ast(file) do
     file
-    |> File.open!()
-    |> IO.read(:all)
+    |> File.read!()
```
Comment (maco) [inline lib/meandro/util.ex:33]: oh, is `:all`/`:eof` the default?

**Review by maco — CHANGES_REQUESTED** [review]

File: lib/meandro/util.ex (hunk 4 above)
Comment (pbrudnick) [reply lib/meandro/util.ex:33]: reads the whole file :)

File: .tool-versions (hunk 3 above)
Comment (pbrudnick) [reply .tool-versions:2]: I did it intended, but I'm open to remove it

Comment (maco) [reply .tool-versions:2]: i'm concerned that when it ends up in the deps directory for other people it'll force them to install those versions

Comment (pbrudnick) [reply .tool-versions:2]: No, this is ignored in the package (like tests and other files)

Comment (maco) [reply .tool-versions:2]: oooooh

**Review by maco — APPROVED** [review]

**Pull request merged**